### PR TITLE
gh-119146: Update `paths` in `jit.yml` 

### PR DESCRIPTION
**Pull request by savannahostrowski**
* Issue: gh-119146

### CODE REVIEW

**Review by FFY00 — APPROVED** [review]
We should actually be able to ignore `Python/perf_jit_trampoline.c` by using a [negative matching pattern](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-and-excluding-branches): `!Python/perf_jit_trampoline.c`.

Other than that, looks good to me.

**Comment by savannahostrowski** [comment]
Oh! I totally didn't know about that. I'll take a look at adding that when I also add .h files to the paths (since I missed adding those in too).

File: .github/workflows/jit.yml [hunk 1]
```diff
@@ -5,11 +5,17 @@ on:
       - '**jit**'
       - 'Python/bytecodes.c'
       - 'Python/optimizer*.c'
+      - '!Python/perf_jit_trampoline.c'
+      - '!Tools/jit/README.md'
+      - '!Tools/jit/mypy.ini'
```
Comment (AlexWaygood) [inline .github/workflows/jit.yml]:
```suggestion
      - '!**/*.md'
      - '!**/*.ini'
```

File: .github/workflows/jit.yml [hunk 2]
```diff
@@ -5,11 +5,17 @@ on:
...
   push:
     paths:
       - '**jit**'
       - 'Python/bytecodes.c'
       - 'Python/optimizer*.c'
+      - '!Python/perf_jit_trampoline.c'
+      - '!Tools/jit/README.md'
+      - '!Tools/jit/mypy.ini'
```
Comment (AlexWaygood) [inline .github/workflows/jit.yml]:
```suggestion
      - '!**/*/README.md'
      - '!**/*/mypy.ini'
```

**Review by AlexWaygood — COMMENTED** [review]
Thanks! Maybe we could just ignore all `.md` and `.ini` files? Then if we added others in the future, we wouldn't have to remember to update this file?

**Comment by savannahostrowski** [comment]
Yeah, that's a fair callout!

**Review by AlexWaygood — APPROVED** [review]
Thanks!

File: .github/workflows/jit.yml [hunk 3]
```diff
@@ -5,11 +5,17 @@ on:
...
+      - '!**/*.md'
+      - '!**/*.ini'
   push:
     paths:
       - '**jit**'
       - 'Python/bytecodes.c'
       - 'Python/optimizer*.c'
+      - '!Python/perf_jit_trampoline.c'
+      - '!**/*/README.md'
+      - '!**/*/mypy.ini'
```
Comment (brandtbucher) [inline .github/workflows/jit.yml]: These should be the same as above, right? Or am I missing something?

```suggestion
      - '!**/*.md'
      - '!**/*.ini'
```

Comment (AlexWaygood) [reply .github/workflows/jit.yml]: Oops, I think that one's on me!

**Comment by hugovk** [comment]
Thanks for the PR!

Shall we backport it, or close the issue?

**Comment by erlend-aasland** [comment]
I think it's ok to backport it to 3.13.

**Pull request merged**